src/adapters/error-handler.js

```diff
@@ -0,0 +1,22 @@
+import { UnknownError } from "../lib/errors"
+
+/**
+ * Handles adapter induced errors.
+ * @param {import("types/adapters").AdapterInstance} adapter
+ * @return {import("types/adapters").AdapterInstance}
+ */
+export default function adapterErrorHandler(adapter) {
+  return Object.keys(adapter).reduce((acc, method) => {
+    const adapterMethod = adapter[method]
+    acc[method] = async (...args) => {
+      try {
+        return await adapterMethod(...args)
```

Comment:
should log out args with debug, and an adapter prefix(?)

@kripod, currently `logger.debug` is rewritten in adapters to include a prefix for  `code`. Do you think we could return a name property next to getAdapter for this purpose?

any thoughts?

Reply:
We may require a `{ displayName: string }` as the minimum required `config` param when instantiating adapters. By doing so, we’ll also solve the typing issue of `Adapter(requiredParam, …)`.

Reply:
After my recent commit, it will be possible for adapters to return a `displayName` property in `getAdapter`. Unfortunately we don't have the adapter configuration when we invoke the `.getAdapter()` method.